module/VuFind/src/VuFind/Content/Covers/Syndetics.php

```diff
@@ -71,46 +83,141 @@ public function __construct($useSSL = false)
-            case 'small':
-                $size = 'SC.GIF';
-                break;
-            case 'medium':
-                $size = 'MC.GIF';
-                break;
-            case 'large':
-                $size = 'LC.JPG';
-                break;
+        $baseUrl = $this->getBaseUrl($key, $ids);
+        if ($baseUrl == false) {
+            return false;
         }
+        if ($this->useSyndeticsCoverImageFallback) {
+            $filename = $this->getImageFilenameFromSize($size);
+            if ($filename == false) {
+                return false;
+            }
+        } else {
+            $xmldoc = $this->getMetadataXML($baseUrl);
+            if ($xmldoc == false) {
+                return false;
+            }
+            $filename = $this->getImageFilenameFromMetadata($xmldoc, $size);
+            if ($filename == false) {
+                return false;
+            }
+        }
+        return $this->getImageUrl($baseUrl, $filename);
+    }
 
+    /**
+     * Return the base Syndetics URL for both the metadata and image URLs.
+     *
+     * @param string $key API key
+     * @param array  $ids Associative array of identifiers (keys may include 'isbn'
+     * pointing to an ISBN object and 'issn' pointing to a string)
+     *
+     * @return string|bool Base URL, or false if no identifier can be used
+     */
+    protected function getBaseUrl($key, $ids)
+    {
         $url = $this->useSSL
             ? 'https://secure.syndetics.com' : 'http://syndetics.com';
-        $url .= '/index.aspx?type=xw12';
+        $url .= "/index.aspx?client={$key}";
+        $ident = '';
         if (isset($ids['isbn']) && $ids['isbn']->isValid()) {
             $isbn = $ids['isbn']->get13();
-            $url .= "&isbn={$isbn}";
-        } else {
-            $isbn = false;
+            $ident .= "&isbn={$isbn}";
         }
         if (isset($ids['issn'])) {
-            $url .= "&issn={$ids['issn']}";
-            $issn = true;
-        } else {
-            $issn = false;
+            $ident .= "&issn={$ids['issn']}";
         }
         if (isset($ids['oclc'])) {
-            $url .= "&oclc={$ids['oclc']}";
-            $oclc = true;
-        } else {
-            $oclc = false;
+            $ident .= "&oclc={$ids['oclc']}";
         }
         if (isset($ids['upc'])) {
-            $url .= "&upc={$ids['upc']}";
-            $upc = true;
-        } else {
-            $upc = false;
+            $ident .= "&upc={$ids['upc']}";
+        }
+        if (empty($ident)) {
+            return false;
+        }
+        return $url . $ident;
+    }
+
+    /**
+     * Calculate the image filename based on the size, without checking if it exists in the metadata.
+     *
+     * @param string $size Size of image to load (small/medium/large)
+     *
+     * @return string|bool Image filename, or false if the size is not 'small', 'medium' or 'large'
+     */
+    protected function getImageFilenameFromSize($size)
+    {
+        switch ($size) {
```

Comment:
Just small suggestion: As VuFind currently needs PHP 8.0 at least, you can utilize match expression instead of switch statement. I think it may be a bit more readable, and is definitely shorter ;) It could look like this:
```
return match ($size) {
    'small' => 'SC.GIF',
    'medium' => 'MC.GIF',
    'large' => 'LC.GIF',
    default => false,
};

```